lib/ios-crash-log.js

```diff
@@ -0,0 +1,63 @@
+import path from 'path';
+
+import _ from 'lodash';
+import _glob from 'glob';
+import { fs, getLogger } from 'appium-support';
+import B from 'bluebird';
+let glob = B.promisify(_glob);
+
+
+require('date-utils');
+let logger = getLogger('IOS-crash-log');
+
+class IOSCrashLog { 
+  constructor() {
+    this.prevLogs = [];
+    this.logsSinceLastRequest = [];
+    this.crashDir = path.resolve(process.env.HOME, 'Library', 'Logs', 'DiagnosticReports');
+  }
+
+  async getCrashes () {
+    try {
+      return glob(`${this.crashDir} /*.crash`);
+    } catch (err) {
+      logger.errorAndThrow('There was a problem getting the crash list');
+      throw err;
+    }
+  }
+
+  async filesToJSON (files) {
+    let logs = [];
```

Comment:
this could be more elegantly written as a map, maybe?

```js
import { asyncmap } from 'asyncbox';
...
return asyncmap(files, async (f) => {
  let filename = path.resolve(this.crashDir, file);
  let stat = await fs.stat(filename);
  return {
    timestamp: stat.ctime,
    level: 'ALL', 
    message: await fs.readFile(filename, 'utf8')
  };
});
```